library/core/src/ptr/mut_ptr.rs

```diff
@@ -112,6 +112,7 @@ impl<T: ?Sized> *mut T {
     /// [`cast_mut`]: #method.cast_mut
     #[stable(feature = "ptr_const_cast", since = "1.65.0")]
     #[rustc_const_stable(feature = "ptr_const_cast", since = "1.65.0")]
+    #[rustc_diagnostic_item = "ptr_cast_const"]
```

Comment:
I think there is high confusion potential between `ptr_const_cast` and `ptr_cast_const`. Maybe this one could be renamed to `ptr_mut_cast_const`?

Reply:
I decided to rename the other one from `ptr_const_cast` to `const_ptr_cast` so that every diagnostic item finishes with the method of it's method.

Reply:
I think that rename improved the situation but the two names are still only distinguished by the order of the components. I think the `mut` should still be added to this one.